### PR TITLE
Milestone 3 - Handling events

### PR DESCRIPTION
**Pull request by cherelemma**
# Implementation Summary

In this Math Magicians project 3rd milestone, the following project requirements are implemented.

- Add calculate and operate modules
- Add constructor and initialize calculator data object
- Add event handler function and state manipulation
- Update style for output bar

## General requirements checklist.

✅  No linters error.
✅  Followed the GitHub flow.
✅  Documented work professionally.

### CODE REVIEW

File: src/components/Calculator.js [hunk 1]
```diff
@@ -1,80 +1,103 @@
...
+          </button>
+          <button type="button" name="8" onClick={onClickEvent}>
+            8
+          </button>
+          <button type="button" name="9" onClick={onClickEvent}>
+            9
+          </button>
+          <button type="button" name="x" onClick={onClickEvent}>
+            X
+          </button>
```
Comment (danmainah) [inline src/components/Calculator.js]: - [ ] @cherelemma kindly check on your multiplication function am getting the error below when trying to multiply 2 numbers
image

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @cherelemma , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ No linter errors
✔️ Gitflow was correctly used
✔️  Good design


# Required Changes ♻️

- [ ] @cherelemma kindly ensure your live demo link is working correctly, it's currently redirecting to your repo instead of the live demo..

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by nahnahdev — APPROVED** [review]
## Project Status: APPROVED :green_circle: 
Hi @cherelemma , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights
- App works as expected :100: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by rloterh — COMMENTED** [review]
Hi Urgaya, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that appears already approved. Otherwise, I suggest that you kindly leave a comment in the PR why you intend resubmitting this approved project.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Pull request merged**